lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/IntegrationExpansion.cxx

```diff
@@ -0,0 +1,195 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief The class building chaos expansions based on a least-squares
+ *         approach. A final cleaning step can be applied.
+ *
+ *  Copyright 2005-2022 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include <cstdlib>
+#include <map>
+
+#include "openturns/IntegrationExpansion.hxx"
+#include "openturns/OSS.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Point.hxx"
+#include "openturns/ResourceMap.hxx"
+#include "openturns/DistributionTransformation.hxx"
+#include "openturns/IdentityFunction.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+typedef Collection<Function> FunctionCollection;
+
+CLASSNAMEINIT(IntegrationExpansion)
+
+static const Factory<IntegrationExpansion> Factory_IntegrationExpansion;
+
+
+/* Default constructor */
+IntegrationExpansion::IntegrationExpansion()
+  : FunctionalChaosAlgorithm()
+{
+  // Nothing to do
+}
+
+
+/* Constructor */
+IntegrationExpansion::IntegrationExpansion(const Sample & inputSample,
+    const Sample & outputSample,
+    const Distribution & distribution)
+  : IntegrationExpansion(inputSample, Point(inputSample.getSize(), 1.0), outputSample, distribution)
```

Comment:
The doc mentions weights to be 1/N ...

Reply:
Fixed

Reply:
Ok, but what is the commit hash and / or why is this discussion unresolved?